include/aws/s3/s3_client.h

```diff
@@ -223,7 +223,7 @@ struct aws_s3_meta_request_options {
      * Note: If set, it will disable compute_content_md5
      */
     /* TODO add functionality for copy object */
-    enum aws_s3_checksum_algorithm checksum_algorithm;
+    enum aws_s3_checksum_algorithm put_checksum_algorithm;
```

Comment:
well, if this will affect copy object in the future, maybe not renaming this to `put_*`